Find Food Trucks SF/Find Food Trucks SF/NearbyViewController.swift

```diff
@@ -45,23 +31,63 @@ class NearbyViewController: UIViewController, UITableViewDataSource, UITableView
                 print("Error fetching food trucks.")
                 return
             }
-
-            for foodTruck in trucks{
-            let coordinate1 = CLLocation(latitude: foodTruck.latitude, longitude: foodTruck.longitude)
-                let distanceInMeters = self.coordinate0.distance(from: coordinate1)
-            if distanceInMeters <= 1609 && !self.nearbyList.contains(foodTruck.name){//find nearby foodtrucks within 1 mile radius, no duplicates.
-                self.nearbyList.append(foodTruck.name)
+            
+            self.foodTrucks = trucks
+            
+            if self.userAllowsLocation() {
+                if let userLocation = self.locationManager.location {
+                    let mile = 1609.34
+                    self.filterTrucks(location: userLocation, distance: mile)
+                }
+            } else {
+                // change the ui to display to a label
+                // asking the user for location permissions
+                // and a button that leads to app settings location services
             }
-        }
+            
             self.myTableView.reloadData()
-            print(self.nearbyList.count)
-        // Do any additional setup after loading the view, typically from a nib.
         }
     }
     
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
+    func userAllowsLocation() -> Bool {
+        return CLLocationManager.locationServicesEnabled() && CLLocationManager.authorizationStatus() == .authorizedWhenInUse
     }
     
+    // loops over all the foodtrucks and filter according to coordinates and distance
+    func filterTrucks(location: CLLocation, distance: Double) {
+        for truck in self.foodTrucks {
+            let truckLocation = CLLocation(latitude: truck.latitude, longitude: truck.longitude)
+            let distanceInMeters = location.distance(from: truckLocation).magnitude
+            
+            if distanceInMeters <= distance {
+                self.nearbyList.append(truck)
+            
+                // TODO: sort the list by distance
+            }
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return nearbyList.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = UITableViewCell(style: .default, reuseIdentifier: nil)
+        cell.textLabel?.text = nearbyList[indexPath.row].name
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        performSegue(withIdentifier: "nearbyToDetailSegueId", sender: self)
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
```

Comment:
good job adding the new segue to show the food truck detail view controller on nearby. Tableview is showing the nearby address and displaying data from our foodtruckviewcontroller when tapped.